### ISSUE TITLE
Calling braintreeInstance.teardown() while 3D-secure modal is open doesn't remove modal's html completely

### ISSUE THREAD

**Issue by mtratsiuk** [braintree/braintree-web-drop-in#463]
### General information

* SDK/Library version: 1.14.1
* Environment: Sandbox
* Browser and OS: Chrome 70.0.3538.77 (64-bit) on Linux 4.18 

### Issue description

Calling `braintreeInstance.teardown()` while 3D-secure modal is open doesn't remove modal's html completely.

### Steps to reproduce
- Create braintree instance:
```
const braintreeInstance = await dropin.create({
        authorization,
        container,
        card: {
          overrides: {
            fields: {
              cvv: {
                type: 'password',
              },
            },
          },
        },
        threeDSecure: {
          amount,
        },
        locale,
      })
```
- Open 3D-secure modal:
```
await braintreeInstance.requestPaymentMethod()
```
- While 3D-secure modal is open call teardown:
```
braintreeInstance.teardown()
```
#### Actual result
3D-secure modal's markup present in the DOM:

![image](https://user-images.githubusercontent.com/9371793/50087424-14500980-0211-11e9-81e2-d534d5785ff8.png)

#### Expected result
3D-secure modal's markup is not present in the DOM

### Additional notes
It looks like `braintreeInstance.teardown()` is aware of 3D-secure modal and tries to [teardown](https://github.com/braintree/braintree-web-drop-in/blob/master/src/dropin.js#L723) it but [ThreeDSecure.prototype.teardown](https://github.com/braintree/braintree-web-drop-in/blob/master/src/lib/three-d-secure.js#L92) doesn't call [ThreeDSecure.prototype._cleanupModal](https://github.com/braintree/braintree-web-drop-in/blob/master/src/lib/three-d-secure.js#L96).

P.S. I've managed to remove 3D-secure modal's markup by calling `braintreeInstance._threeDSecure.cancel()` before `braintreeInstance.teardown()`, but I believe it shoudn't be necessary

**Comment by crookedneighbor** [comment]
Thanks for the report, we'll get this fixed up.

**Comment by crookedneighbor** [comment]
This is fixed in v1.15.0
